internal/app/networkd/pkg/nic/nic.go

```diff
@@ -312,18 +312,26 @@ func (n *NetworkInterface) Renew() {
 // address TTL. If that fails, we'll continue to attempt to retry every
 // halflife.
 func (n *NetworkInterface) renew(method address.Addressing) {
+	const minRenewDuration = 5 * time.Second // protect from renewing too often
+
 	renewDuration := method.TTL() / 2
 
 	var err error
 
 	for {
-		<-time.After(renewDuration)
+		time.Sleep(renewDuration)
```

Comment:
`<-time.After()` involves a channel which should be GC'ed, much better to use `time.Sleep()` here